libs/wingc/src/capture.rs

```diff
@@ -328,8 +328,10 @@ fn scan_captures_in_inflight_scope(scope: &Scope, diagnostics: &mut Diagnostics)
 	let env_ref = scope.env.borrow();
 	let env = env_ref.as_ref().unwrap();
 
-	// Make sure we're looking for captures only in inflight code
-	assert!(matches!(env.phase, Phase::Inflight));
+	// Early bail-out if we're looking for captures in preflight code - an error will be reported in type checking.
+	if matches!(env.phase, Phase::Preflight) {
```

Comment:
After emitting an error instead of the assert in type_check.rs - this assert surfaced. Decided to bail early if calling the `scan_captures_in_inflight_scope` method in preflight. Happy to hear if we should further limit this if statement.